src/term.rs

```diff
@@ -157,19 +170,36 @@ impl Term {
         })
     }
 
+    /// Return a terminal for the given Read/Write pair.
+    pub fn read_write_pair<R, W>(read: R, write: W) -> Term
+    where
+        R: Read + fmt::Debug + AsRawFd + Send + 'static,
+        W: Write + fmt::Debug + AsRawFd + Send + 'static,
+    {
+        Term::with_inner(TermInner {
+            target: TermTarget::ReadWritePair(ReadWritePair {
+                read: Arc::new(Mutex::new(read)),
+                write: Arc::new(Mutex::new(write)),
+            }),
+            buffer: None,
+        })
+    }
+
     /// Returns the style for the term
     #[inline]
     pub fn style(&self) -> Style {
-        match self.target() {
+        match self.inner.target {
             TermTarget::Stderr => Style::new().for_stderr(),
             TermTarget::Stdout => Style::new().for_stdout(),
+            // TODO: Needs an implementation. But also: not Copy.
+            TermTarget::ReadWritePair(_) => Style::new().for_stderr(),
```

Comment:
`Style` is not `Copy`, so this method is creating a new one for each call. Should probably return a reference.

Additionally, there is a question of whether someone constructing an arbitrary `Term` would want to be able to `Style` it explicitly.